src/NServiceBus.AcceptanceTests/SelfVerifications/EndpointNameEnforcementTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.SelfVerification
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using AcceptanceTesting;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class EndpointNameEnforcementTests
```

Comment:
Is there any value in running this in the downstreams? I mean it will already be to late at that point since the core already shipped tests that will fail due to long names.

I propose we move this to [the core folder to just have it be executed when the core runs its tests](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj#L40).

That means we can  hardcode the length to 77 as well.

Thoughts?

Reply:
I agree. We only need to protect the core tests since those get shipped to downstreams.

If a downstream project adds a test, they'll immediately know if they hit a length problem and can adjust the the test directly.

That means we don't need any of the TestSuiteConstraints modifications.